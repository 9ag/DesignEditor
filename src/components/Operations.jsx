import React from 'react'
class Operations extends React.Component {
  constructor(props) {
    super(props)
  }
  render() {
    const {onSearch, onAdvance, onClear} = this.props
    return (
      <div className="search-group">
        <button type="button" onClick={onSearch} id="btn-search" className="btn btn-primary btn-long btn-sm mr-8">
          <i className="fa fa-search"></i>
        </button>
        <button onClick={onAdvance} type="button" className="btn btn-sm btn-primary mr-8" data-toggle="modal" data-target="#advanSearch" data-bind="css: {'btn-magenta': isAdvanceSearch, 'btn-primary': !isAdvanceSearch()}">
          <i className="fa fa-bars"></i>
        </button>
        <div className="btn-group" id="filterContainer">
          <button type="button" className="btn btn-primary btn-sm dropdown-toggle" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false"
            name="OpenFilterContainerBtn" data-bind="click: filterContext.filterListContext.render.bind(filterContext.filterListContext)">
            <i className="fa fa-filter"></i>
          </button>
          <div id="management_filterContainer">
            <div id="filter_management_model_management_filterContainer" style={{display: 'none'}} data-bind="with: filterContext.filterListContext.filterManagementContext">
            </div>
          </div>
        </div>
        {/* <button type="button" onClick={onClear} name="AdvClearFilerSearchBtn" class="btn btn-default btn-sm m0">
          <i class="fa fa-undo"></i>
        </button> */}
      </div>
    )
  }
}

export default Operations