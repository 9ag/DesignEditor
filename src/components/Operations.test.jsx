
import { shallow, mount, render } from 'enzyme';
import React from 'react';
import renderer from 'react-test-renderer';

import * as enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
enzyme.configure({ adapter: new Adapter() });

import Operations from './Operations';
describe("Operations component test", () => {
    let onAdvance = null;
    let onSearch = null;
    beforeEach(() => {
      onAdvance = jest.fn();
      onSearch = jest.fn();
    });

    it("click callback test", () => {
      const callback = jest.fn();
      const component = shallow(<Operations onSearch={onSearch} onAdvance={onAdvance} />);
      component.find('button').at(0).simulate('click');
      expect(onSearch.mock.calls.length).toBe(1);
      component.find('button').at(1).simulate('click');
      expect(onAdvance.mock.calls.length).toBe(1);
    });
     
    it('snapshot test', () => {
      const component = renderer.create(<Operations onSearch={onSearch} onAdvance={onAdvance} />).toJSON();
      expect(component).toMatchSnapshot();
    });
});