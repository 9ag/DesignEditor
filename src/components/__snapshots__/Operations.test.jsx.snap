// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Operations component test snapshot test 1`] = `
<div
  className="search-group"
>
  <button
    className="btn btn-primary btn-long btn-sm mr-8"
    id="btn-search"
    onClick={[MockFunction]}
    type="button"
  >
    <i
      className="fa fa-search"
    />
  </button>
  <button
    className="btn btn-sm btn-primary mr-8"
    data-bind="css: {'btn-magenta': isAdvanceSearch, 'btn-primary': !isAdvanceSearch()}"
    data-target="#advanSearch"
    data-toggle="modal"
    onClick={[MockFunction]}
    type="button"
  >
    <i
      className="fa fa-bars"
    />
  </button>
  <div
    className="btn-group"
    id="filterContainer"
  >
    <button
      aria-expanded="false"
      aria-haspopup="true"
      className="btn btn-primary btn-sm dropdown-toggle"
      data-bind="click: filterContext.filterListContext.render.bind(filterContext.filterListContext)"
      data-toggle="dropdown"
      name="OpenFilterContainerBtn"
      type="button"
    >
      <i
        className="fa fa-filter"
      />
    </button>
    <div
      id="management_filterContainer"
    >
      <div
        data-bind="with: filterContext.filterListContext.filterManagementContext"
        id="filter_management_model_management_filterContainer"
        style={
          Object {
            "display": "none",
          }
        }
      />
    </div>
  </div>
</div>
`;
